pkg/provider/azure.go

```diff
@@ -1220,3 +1230,12 @@ func (az *Cloud) ShouldNodeExcludedFromLoadBalancer(nodeName string) (bool, erro
 
 	return az.excludeLoadBalancerNodes.Has(nodeName), nil
 }
+
+func isNodeReady(node *v1.Node) bool {
+	for _, cond := range node.Status.Conditions {
+		if cond.Type == v1.NodeReady && cond.Status == v1.ConditionTrue {
```

Comment:
Could we also check whether this is a new node or not here (via "node.cloudprovider.kubernetes.io/uninitialized")? new node should not be removed so as to reduce number of ARM calls. 

https://github.com/kubernetes/cloud-provider/blob/5a236942a3ca6f0454bfcb979cc31207af4894b2/api/well_known_taints.go#L20-L24

Reply:
Sure! I added a check in lines 1100 and 1054. Thanks!